### DISCUSSION TITLE
Pager wrapper working with man but not with nnn help

### DISCUSSION THREAD

**Discussion by ghost** [jarun/nnn#1697] [Q&A] [answered]
In order to handle the [different ways in which man pages and my keyboard encode hyphens](https://github.com/walles/moar/issues/142), I created the following wrapper around my pager:

```
#!/usr/bin/env bash
printf "%s" "$(cat)" | sed $'s/\u2010/\u002d/g' | moar
```

And I set it to the "PAGER" env variable.

It is working correctly with man pages, e.g.:
![image](https://github.com/jarun/nnn/assets/118444583/e1fa1899-ebee-47ac-9567-4bf104881236)

But not with nnn help (`?`). This is what is shown:
![image](https://github.com/jarun/nnn/assets/118444583/587ed0b1-0e49-47c3-b13a-36da24d8b439)

It's like is passing empty page.

Of course nnn help works correctly with bare moar, or bare less for that matter.

Do you know why?

**Comment by N-R-K** [comment, marked as answer]
1. `printf "%s" "$(cat)"` sed should be able to read from stdin just fine, so this looks entirely useless
2. Your wrapper doesn't seem to handle files passed via argument

change it to `sed $'s/\u2010/\u002d/g' "$@" | moar` and see if that works.

**Comment by ghost** [comment]
It does work.
Altough I'm not understanding the behavior: sed reads from stdin, and if nothing is in std, it reads from command line arguments?

EDIT: Yes, it's the other way around.
From sed's manpage:
> All remaining arguments are names of input files; if no input files are specified, then the standard input is read.

So `nnn` is passing its help as command line argument it seems?

**Reply by N-R-K** [reply]
Yes, *most* unix utilities behave that way (with certain exceptions such as `tr`). If you give it a file then it'll use the file - so `cat`ting things into something is usually a [useless use of cat](https://en.wikipedia.org/wiki/Cat_(Unix)#Useless_use_of_cat).

**Reply by ghost** [reply]
Thanks! I learnt a new thing!
